Docs: Expand ruler recommended alerting docs (#1719)
thanos-io/thanos: docs/components/rule.md

indicate connection, incompatibility or misconfiguration problems.

* `prometheus_rule_evaluation_failures_total`. If greater than 0, it means that that rule failed to be evaluated, which results in

> either gap in rule or potentially ignored alert. This metrics indicates an unhealthy downstream query component. Alert heavily on this if this happens for longer than your alert thresholds.

Reviewer: ```suggestion
either gap in rule or potentially ignored alert. This metric might indicate problems on the queryAPI endpoint you use.

Alert heavily on this if this happens for longer than your alert thresholds.
```

Author: SGTM!

indicate connection, incompatibility or misconfiguration problems.

* `prometheus_rule_evaluation_failures_total`. If greater than 0, it means that that rule failed to be evaluated, which results in
either gap in rule or potentially ignored alert. This metric might indicate problems on the queryAPI endpoint you use.

> 

Reviewer 2: This new line here breaks the formatting of the list. This makes the line “alert heavily...” appear out of the list, and then the list continues.
